packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,721 @@
+[`use`](https://markdown-it.github.io/markdown-it/#MarkdownIt.use) method and identifying the value of the
+`mention-url` that matches the `display-name`. The `mention-url` and `name` will then be generated as an object from the
+`nimble-mapping-mention-user` elements. This custom node will be added `before` the `autolink` mark to give
+the highest precedence to the `mention` node.
+
+If the user is no longer a valid user or the mapping elements were not yet updated but the markdown string stores a valid
+user URL that matches the pattern, then the `@mention` node will parse as a user ID, instead of a user name as they are not
+mapped with any mapping elements.
+
+#### 4. _Defining node in markdown serializer_:
+
+The rendered `@mention` node will be constructed into a markdown string by extracting the `mention-url` from
+the `span` element in the editor when the `getMarkdown()` method is called.
+
+The example markdown string constructed for the below DOM element rendered in the editor is `<https://users/1234-5678>`.
+
+```HTML
+<nimble-rich-text-user-mention-view
+    mention-type="mention"
+    mention-url="https://users/1234-5678"
+    mention-label="Mary"
+    contenteditable="false"
+    >@Mary</nimble-rich-text-user-mention-view
+>
+```
+
+The `mentioned-users` read-only property, as described in the [API section of the editor](./README.md#api), will undergo an update within the
+mentionNode. This update involves extracting the user ID from the URL that matches the pattern specified in the configuration element and
+then adding these user IDs to an array.
+
+#### 5. _nimble-rich-text-mention-users_:
+
+An abstract base class, `RichTextMention`, is defined as the parent for all elements that contain mentions, and it will possess the following properties:
+
+1. `character`: string - is a specific symbol to trigger the mention popup. For user mention, it is **"@"**.
+2. `icon`: string - element name of the icon for the corresponding toolbar button.
+
+The `nimble-rich-text-mention-users` is a subclass derived from the base class and provides the essential values for the properties mentioned above,
+specifically tailored for user mentions. These values are kept as part of the `MentionInternals` and will be utilized by various components
+that require access to these specific values.
+
+Validation will be integrated into the internal workings through a `validConfiguration` flag. There is a base class called `RichTextMentionValidator`
+responsible for handling validation, with methods for setting and retrieving the validation configuration. This approach ensures that the state of
+valid and invalid values passed within the mapping element is effectively managed. The class obtains the name of the validity flag to communicate this
+information via a public API called `validity`.
+
+By deriving from the base, the mention options can validate the following conditions for the `mention-url` and `display-name` values in mapping element:
```

Comment:
I'd expect all mentions to have an `href` defined and several of these validations (unique href, href not missing, href valid, pattern valid) to be handled in a base class shared by all mentions

Reply:
Agreed. Updated the same.